src/monodroid/jni/platform-compat.hh

```diff
@@ -0,0 +1,66 @@
+#ifndef __PLATFORM_COMPAT_HH
+#define __PLATFORM_COMPAT_HH
+
+#if __cplusplus >= 201703L
+#define UNUSED_ARG [[maybe_unused]]
+#else
```

Comment:
Please, as per convention elsewhere in `src/monodroid/jni`, include comments on `#else` and `#endif` so that mere mortals (read: me) can more easily determine what a given block is *for*.